salt/modules/openvswitch.py

```diff
@@ -0,0 +1,302 @@
+        List or False.
+    """
+    if retcode == 0:
+        ret = stdout.split(splitstring)
+        return ret
+    else:
+        return False
+
+
+def bridge_list():
+    """
+    Lists all existing real and fake bridges.
+
+    Returns:
+        List of bridges (or empty list), False on failure.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' openvswitch.bridge_list
+    """
+    cmd = _detect_os() + ' list-br'
+    result = __salt__['cmd.run_all'](cmd)
+    retcode = result['retcode']
+    stdout = result['stdout']
+    return _stdout_list_split(retcode, stdout)
+
+
+def bridge_exists(br):
+    """
+    Tests whether bridge exists as a real or fake  bridge.
+
+    Returns:
+        True if Bridge exists, else False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' openvswitch.bridge_exists br0
+    """
+    cmd = _detect_os() + ' br-exists {0}'.format(br)
+    result = __salt__['cmd.run_all'](cmd)
+    retcode = result['retcode']
+    return _retcode_to_bool(retcode)
+
+
+def bridge_create(br, may_exist=True):
+    """
+    Creates a new bridge.
+
+    Args:
+        br: A string - bridge name
+        may_exist: Bool, if False - attempting to create a bridge that exists returns False.
+
+    Returns:
+        True on success, else False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' openvswitch.bridge_create br0
+    """
+    param_may_exist = _param_may_exist(may_exist)
+    cmd = _detect_os() + ' {1}add-br {0}'.format(br, param_may_exist)
+    result = __salt__['cmd.run_all'](cmd)
+    return _retcode_to_bool(result['retcode'])
+
+
+def bridge_delete(br, if_exists=True):
+    """
+    Deletes bridge and all of  its  ports.
+
+    Args:
+        br: A string - bridge name
+        if_exists: Bool, if False - attempting to delete a bridge that does not exist returns False.
+
+    Returns:
+        True on success, else False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' openvswitch.bridge_delete br0
+    """
+    param_if_exists = _param_if_exists(if_exists)
+    cmd = _detect_os() + ' {1}del-br {0}'.format(br, param_if_exists)
+    result = __salt__['cmd.run_all'](cmd)
+    retcode = result['retcode']
+    return _retcode_to_bool(retcode)
+
+
+def port_add(br, port, may_exist=False):
+    """
+    Creates on bridge a new port named port.
+
+    Returns:
+        True on success, else False.
+
+    Args:
+        br: A string - bridge name
+        port: A string - port name
+        may_exist: Bool, if False - attempting to create a port that exists returns False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' openvswitch.port_add br0 8080
+    """
+    param_may_exist = _param_may_exist(may_exist)
+    cmd = _detect_os() + ' {2}add-port {0} {1}'.format(br, port, param_may_exist)
+    result = __salt__['cmd.run_all'](cmd)
+    retcode = result['retcode']
+    return _retcode_to_bool(retcode)
+
+
+def port_remove(*args, **kwargs):
```

Comment:
Is there a reason we can't explicitely declare args here? We prefer not to have *args and **kwargs if at all possible.